### PR TITLE
docs/migration: add prereqs, align guides for all languages

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:**
- docs/building-operators/*/migration.md: make sure these are aligned and have prerequisites

**Motivation for the change:** migration guides should all have prerequisites. Follows up on #4665.

/kind documentation

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [x] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: website/content/en/docs/building-operators/helm/migration.md [hunk 1]
```diff
@@ -149,22 +160,33 @@ Note that the following environment variables are no longer used.
...
 
 The default port used by the metric endpoint binds to was changed from `:8383` to `:8080`. To continue using port `8383`, specify `--metrics-addr=:8383` when you start the operator.
 
-## Checking the changes
+### Verify the migration
+
+The project can now be deployed on cluster by running the command:
+
+```sh
+make deploy IMG=example.com/memcached-operator:v0.0.1
```
Comment (varshaprasad96) [inline website/content/en/docs/building-operators/helm/migration.md:175]: For the helm-operator, we are initially guiding users to scaffold nginx-operator (https://sdk.operatorframework.io/docs/building-operators/helm/migration/#creating-a-new-project) in the migration guide. Can we modify this to nginx to have uniformity?

Comment (camilamacedo86) [reply website/content/en/docs/building-operators/helm/migration.md:175]: IHMO: We need update here for nginx-operator as @varshaprasad96 suggested
And then, at the end of the doc put a note to describe that the users are able to check a final result example in the testdata of the Memcached scenario which is the same example adapted across the docs.

**Review by camilamacedo86 — APPROVED** [review]
Just a comment to be addressed https://github.com/operator-framework/operator-sdk/pull/4683/files. After that it shows great for me 👍

File: website/content/en/docs/building-operators/helm/migration.md (hunk 1 above)
Comment (estroz) [reply website/content/en/docs/building-operators/helm/migration.md:175]: If you look through all the helm docs, you'll notice a mix of memcached and nginx. I'll change this here, but these docs need to align on GVK.

Comment (estroz) [reply website/content/en/docs/building-operators/helm/migration.md:175]: The helm testdata really should align with docs. I don't see a reason not to use memcached in the helm example, over nginx. Thoughts?

**Pull request merged**